r2-core/src/main/java/com/linkedin/r2/transport/http/client/SmoothRateLimiter.java

```diff
@@ -78,19 +97,26 @@ public SmoothRateLimiter(ScheduledExecutorService scheduler, Executor executor,
     _executor = executor;
     _pendingCallbacks = pendingCallbacks;
     _maxBuffered = maxBuffered;
+    _bufferOverflowMode = bufferOverflowMode;
+    _rateLimiterName = rateLimiterName;
```

Comment:
What is the rateLimiterName used for? Do we expect CapacityAwareClient to pass the name?

Reply:
Yes, it is used for logging purposes. I realized that it is very useful to know it when you throw the exception or log the warning, since otherwise you don't know the queue it was happening.
e.g. you have several services that are using the queue, but only one is the offender. How do you identify it?

It is the same concept as the threadName, without it you're completely lost at debug time